atomic_reactor/util.py

```diff
@@ -808,11 +809,59 @@ def query_registry(registry_session, image, digest=None, version='v1', is_blob=F
     logger.debug("query_registry: querying {}, headers: {}".format(url, headers))
 
     response = registry_session.get(url, headers=headers)
+    for r in chain(reversed(response.history), [response]):
```

Comment:
Since `history` "is sorted from the oldest to the most recent request", does it make sense to first look at response, then through the reversed history list? Alternatively, first look at non-reversed history list and then response. The current iteration order is not sequential.

Reply:
Good catch! Don't know how I managed to read the documentation wrong there.